code/game/objects/items.dm

```diff
@@ -1054,6 +1055,34 @@ var/global/list/image/blood_overlays = list()
 
 	blood_overlays[type] = image(I)
 
+/obj/item/apply_luminol()
+	if(!..())
+		return FALSE
+	if(!blood_overlays[type]) //Blood overlay generation if it lacks one.
+		generate_blood_overlay()
+	if(blood_overlay != null)
```

Comment:
`if(blood_overlay)` is more general. This will fail if `blood_overlay` is zero or an empty string.

Reply:
Resolved.